src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,61 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.DukeSideEffectException;
+import duke.io.FileIO;
+import duke.io.Printer;
+import duke.task.TaskList;
+import java.io.IOException;
+
+public abstract class Command {
+  public static final String LIST = "list";
+  public static final String MARK = "mark";
+  public static final String UNMARK = "unmark";
+  public static final String TODO = "todo";
+  public static final String DEADLINE = "deadline";
+  public static final String EVENT = "event";
+  public static final String DELETE = "delete";
+	public static final String FIND = "find";
+
+  protected Printer out;
+  protected TaskList taskList;
+  private FileIO savefile;
+
+  private boolean saveAfterAction = false;
```

Comment:
Perhaps it would be better to change the name to include a prefix like is? A possible change could be isSavedAfterAction. This could be easier to understand that it is a boolean.